docs/Syntax.ebnf

```diff
@@ -1,62 +1,49 @@
-Comment ::= SingleLineComment | MultilineComment
-
-Literal ::= StringLiteral | NumberLiteral | Boolean
-
-DecStatement ::= (VarKeyword | LetKeyword) Symbol (Colon Type)? (Equal Expr)?
-
-CodeBlock ::= OpenCurlyBracket grammar* CloseCurlyBracket
-
-Invoke ::= Symbol OpenParenthesis (Expr (Comma Expr)*)? CloseParenthesis
-
-InfixOperator ::= Equal
-                | Plus | Minus | Mul | Divide
-                | Or | And | Xor
-                | CompareEqual | NotEqual
-                | GreaterThan | LessThan
-                | GreaterThanEqual | LessThanEqual
-
-PrefixOperator ::= Not | Minus | MinusMinus | PlusPlus
-
-SuffixOperator ::= MinusMinus | PlusPlus
-
-Expr ::= OpenParenthesis Expr CloseParenthesis
-       | Literal | Symbol
-       | Invoke
-       | CodeBlock
-       | Array
-       | Fn
-       | Expr (InfixOperator | Symbol) Expr
-       | (Symbol | PrefixOperator) Expr
-       | Expr (Symbol | SuffixOperator)
-
-Params ::= OpenParenthesis (Symbol Colon Type (Comma Symbol Colon Type)*)? CloseParenthesis
-Fn ::= FnKeyword Symbol Params (Colon Type)? CodeBlock
-
-Type ::= Symbol (OpenSquareBracket CloseSquareBracket)*
-
-Array ::= OpenSquareBracket (Expr (Comma Expr)* Comma?)? CloseSquareBracket
-
-ForeachLoop ::= Symbol InKeyword Expr
-Loop ::= LoopKeyword OpenParenthesis (ForeachLoop | Expr) CloseParenthesis CodeBlock
-
-If ::= IfKeyword OpenParenthesis Expr CloseParenthesis CodeBlock
-       (ElseKeyword (CodeBlock | If))?
-
-Return ::= ReturnKeyword Expr
-
-Struct ::= StructKeyword Symbol OpenCurlyBracket (Symbol Colon Type)* CloseCurlyBracket
-
-Impl ::= ImplKeyword Symbol CodeBlock
-
-Attribute ::= At Invoke
-
-AffixFn ::= (InfixKeyword | PrefixKeyword | SuffixKeyword)
-            (Fn | OpKeyword AffixOperator Params (Colon Type)? CodeBlock)
-
-grammar ::= (Comment
-             | DecStatement
-             | Expr
-             | Return
-             | Struct
-             | Impl
-             | Attribute)*
+alphaChar = ? all alphabetic characters ? | "_";
+numChar = ? all numeric characters ?;
+
+identifier = alphaChar, {alphaChar | numChar}; (* Valid identifier *)
+number = ["-"], (numChar - "0"), {numChar}, [".", numChar*]; (* Decimal number *)
+operation =
+("!" | "#" | "$" | "%" | "&" | "*" | "+" | "-" | "/" | ":" | "<" | "=" | ">" | "?" | "@" | "^" | "|" | "~")* - (* Buildup of operators *)
+("=" | "//" | "/*" | "*/") (* Invalid or special *)
+; (* all operators *)
+
+type = identifier; (* Any type *)
+parameter = identifier + ":", type; (* Parameter to function *)
+
+comment = ("//", ? all characters ?, "\n") | ("/*", ? all characters ?, "*/");
+
+vardef = ("var" | "let"), identifier; (* Beginning of variable definition *)
+funcdef = "fn", identifier, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
+opdef = "infix", "op", operator, "(", parameter, {",", parameter}, ")"; (* Beginning of function definition *)
```

Comment:
infix isn't the only option, there also prefix and suffix and you can put the affixes on functions as well